android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityAndroidPayFragment.java

```diff
@@ -124,23 +125,48 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         PayHelper.handleWalletResponse(requestCode, resultCode, data, new PayHelper.WalletResponseHandler() {
 
             @Override
-            public void onMaskedWallet(MaskedWallet maskedWallet) {
+            public void onMaskedWallet(final MaskedWallet maskedWallet) {
                 super.onMaskedWallet(maskedWallet);
 
-                // TODO: Send MaskedWallet information back to Unity to update that state
-                UnityAndroidPayFragment.this.maskedWallet = maskedWallet;
-
-                // TODO: Got a wallet - show the confirmation fragment
+                MailingAddressInput input = new MailingAddressInput(maskedWallet.getBuyerShippingAddress());
+                UnityMessage msg = UnityMessage.fromAndroid(input.toJsonString());
+                MessageCenter.MessageReceiver receiver = new MessageCenter.MessageReceiver(
+                    unityDelegateObjectName,
+                    MessageCenter.Method.ON_UPDATE_SHIPPING_ADDRESS
+                );
+
+                MessageCenter.sendMessageTo(msg, receiver, new MessageCenter.MessageCallbacks() {
+                    @Override
+                    public void onResponse(String jsonResult) {
+                        UnityAndroidPayFragment.this.maskedWallet = maskedWallet;
```

Comment:
Here can be a mem leak that I mentioned above, so if unity won't reply on message, the callback that we created will be always in memory and this callback keep strong reference to activity (UnityAndroidPayFragment.this will create implicit internal reference)